msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -66,10 +107,100 @@ private void initializeSharedPreferenceFileManager(@NonNull final Context contex
                 new StorageHelper(context));
     }
 
+
     @Override
-    public void getCurrentAccount(@NonNull final CurrentAccountCallback callback) {
+    public void getCurrentAccount(final CurrentAccountCallback callback) {
+        final String methodName = ":getCurrentAccount";
         final PublicClientApplicationConfiguration configuration = getConfiguration();
 
+
+        try {
+            if (MSALControllerFactory.brokerEligible(
+                    configuration.getAppContext(),
+                    configuration.getDefaultAuthority(),
+                    configuration) && mIsSharedDevice) {
+                getCurrentAccountFromSharedDevice(callback, configuration);
+                return;
+            }
+
+            ApiDispatcher.initializeDiagnosticContext();
+
+            com.microsoft.identity.common.internal.logging.Logger.verbose(
+                    TAG + methodName,
+                    "Getting the current account"
+            );
+
+
+            final OperationParameters params = OperationParametersAdapter.createOperationParameters(mPublicClientConfiguration);
+            final LoadAccountCommand command = new LoadAccountCommand(
+                    params,
+                    MSALControllerFactory.getAcquireTokenController(
+                            mPublicClientConfiguration.getAppContext(),
+                            params.getAuthority(),
+                            mPublicClientConfiguration
+                    ),
+                    new TaskCompletedCallbackWithError<List<ICacheRecord>, Exception>() {
+                        @Override
+                        public void onTaskCompleted(final List<ICacheRecord> result) {
+                            if (null == result || result.size() == 0) {
+                                com.microsoft.identity.common.internal.logging.Logger.verbose(
+                                        TAG + methodName,
+                                        "No account found.");
+                                checkCurrentAccountNotifyCallback(callback, null);
+                            } else {
+                                // First, transform the result into IAccount + TenantProfile form
+                                final List<IAccount>
+                                        accounts = AccountAdapter.adapt(result);
+
+                                final String trimmedIdentifier = getPersistedCurrentAccount().getHomeAccountId();
```

Comment:
getPersistedCurrentAccount() could be nil.

Reply:
Agreed.